sktime/classification/kernel_based/_rocket_classifier.py

```diff
@@ -49,13 +47,12 @@ class ROCKETClassifier(BaseClassifier):
     tsml/classifiers/shapelet_based/ROCKETClassifier.java
     """
 
-    # Capability tags
-    capabilities = {
-        "multivariate": True,
-        "unequal_length": False,
-        "missing_values": False,
-        "train_estimate": False,
-        "contractable": False,
+    _tags = {
+        "capability:multivariate": True,
+        "capability:unequal_length": False,
+        "capability:missing_values": False,
+        "capability:train_estimate": False,
+        "capability:contractable": False,
     }
```

Comment:
I'd also put these in the base class with the following defaults:
```python
        "capability:multivariate": False,
        "capability:unequal_length": False,
        "capability:missing_values": False,
        "capability:train_estimate": False,
        "capability:contractable": False,
```
So here you would only have to put

```python
    _tags = {
        "capability:multivariate": True
    }
```

because that's the only one that would differ from the default - but no harm having all of them in the concrete class.

Reply:
I don't really like such implicit defaults, custom tag inheritance + defaults make things difficult. When all tags are listed in one class, it is okay, I can get it from the code. But if some tags are in base class, some in subclass - the only method to get tag values is to instantiate object and inspect it in runtime

Reply:
>I don't really like such implicit defaults, custom tag inheritance + defaults make things difficult. When all tags are listed in one class, it is okay, I can get it from the code. But if some tags are in base class, some in subclass - the only method to get tag values is to instantiate object and inspect it in runtime

I get your point, and agree with it.

Reply:
In fact, my opinion (in previous core developer discussions) was this:

* have default tags in the base class (not `BaseEstimator` but of the scitype, e.g., `BaseClassifier`), so we have "safe defaults" in case someone forgets
* also ask extenders to fill in all the tags in any concrete subclass, as a kind of "poka yoke" which makes it less likely that people forget to enter tags, and since it's easier to look at the code and see all the tags.

The consensus we ended up with though, was:

* have defaults in base classes
* do *not* ask to enter the tags which are equal to defaults

I don't like that entirely, but that's what we arrived at.

Reply:
in any case, I wouldn't see it as a PR blocker if you fill in all the tags in the subclass.

Reply:
well I was not in favour of custom tag inheritance, hence why I set up capabilities as a much simpler system. Personally I much prefer boolean tags to freeform strings. Sorry it be awkward, but these changes are going to effect every single classifier, so I would like a proper discussion of this rather than just slipping it through with a single classifier. We can refactor later if it is collectively decided to be desirable, but for now I would like a simple porting over to the _tags, and defaults in the base class as Franz suggests.  contractable can change to train-time-contractable if you want.

Reply:
> In fact, my opinion (in previous core developer discussions) was this:
> 
> * have default tags in the base class (not `BaseEstimator` but of the scitype, e.g., `BaseClassifier`), so we have "safe defaults" in case someone forgets
> * also ask extenders to fill in all the tags in any concrete subclass, as a kind of "poka yoke" which makes it less likely that people forget to enter tags, and since it's easier to look at the code and see all the tags.
> 
> The consensus we ended up with though, was:
> 
> * have defaults in base classes
> * do _not_ ask to enter the tags which are equal to defaults
> 
> I don't like that entirely, but that's what we arrived at.

Oh ok. Having no defaults and also not insisting on assigning values will increase the chances of run time errors. My view would be to have scitype defaults but not insist on subclasses assigning a value, I view that as bloaty.  One point of inheritance is to provide the default behaviour or explicitly specialise. But as I say, this PR does not need to decide the whole issue, I suggest we just get this in with the minimal refactoring and then consider from there.

Reply:
I see there is still no agreement about defaults and overrides, and I agree with Tony we should keep things simple until there is some decision on this question.

So for now lets only rename capabilities to tags with `capability:` prefix, as it is very simple for contributors. If later we decide to add tags defaults to base class and enforce to remove from concrete classes tags that have default values, we can do it in automated way at any stage of refactoring or after it.

Reply:
I understand there is already an agreement to have default tag values, but it is not still clear where to keep them, in `BaseEstimator` or `BaseClassifier` and should concrete class list all tag values or only distinct from defaults. I guess when this decision is made, it should be put down in TSC template there https://github.com/alan-turing-institute/sktime/blob/84c4ff004b7e6735becc0e63f105709e2e0b3e8a/extension_templates/classification.py#L70

For now, lets just convert capabilities into tags.